allennlp/tests/common/logging_test.py

```diff
@@ -0,0 +1,58 @@
+import os
+import logging
+from allennlp.common.logging import AllenNlpLogger
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestLogging(AllenNlpTestCase):
```

Comment:
Can you add a test that makes sure that this sequence only produces one log entry:
```Python
logger.debug_once("There are %d lights.", 4)
logger.debug_once("There are %d lights.", 5)
```
The reason I am concerned about it is that if this doesn't work, we might end up storing millions of messages in `self.msg`.

Reply:
In fact, maybe add that to the test, i.e., make sure that after that sequence, `sizeof(logger.msg)` is 1.